agents/appmotion_motor/appmotion_motors_agent.py

```diff
@@ -0,0 +1,1142 @@
+                    "motor1_connection": 1,
+                    "motor1_move_status": False,
+                }
+            }
+
+        Parameters:
+            motor (int): Determines which motor, either 1 or 2, 3 is for all
+                    motors. (default 3)
+            verbose (bool): Prints output from motor requests if True.
+                (default False)
+            f_sample (float): Sampling rate in Hz. (default 2)
+        """
+        if params is None:
+            params = {}
+
+        pm = Pacemaker(params['f_sample'], quantize=True)
+
+        if not self.initialized:
+            raise Exception("Motors are not initialized")
+
+        with self.lock.acquire_timeout(timeout=0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn(
+                    "Could not start acq because {} is already running".format(
+                        self.lock.job))
+                return False, "Could not acquire lock."
+            self.log.info(
+                f"Starting data acquisition for stages at {params['f_sample']} Hz")
+            session.set_status('running')
+            self.take_data = True
+            last_release = time.time()
+
+            mot_list = [self.motor1, self.motor2]
+            while self.take_data:
+                if time.time() - last_release > 1.:
+                    if not self.lock.release_and_acquire(timeout=10):
+                        self.log.warn(
+                            f"Could not re-acquire lock now held by {self.lock.job}.")
+                        return False, "could not re-acquire lock"
+                    last_release = time.time()
+                pm.sleep()
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'positions',
+                    'data': {}}
+
+                for mot in mot_list:
+                    if mot.mot_id is not None:
+                        try:
+                            self.log.debug(
+                                f"getting position/move status of motor{params['motor']}")
+                            self.move_status = mot.is_moving(session, params)
+                            pos = mot.get_position_in_inches()
+                            if self.move_status:
+                                data['data'][f'motor{mot}_encoder'] = -1
+                            else:
+                                e_pos = mot.retrieve_encoder_info()
+                                data['data'][f'motor{params["motor"]}_encoder'] = e_pos[0]
+                            data['data'][f'motor{params["motor"]}_stepper'] = pos[0]
+                            data['data'][f'motor{params["motor"]}_connection'] = 1
+                            data['data'][f'motor{params["motor"]}_move_status'] = self.move_status
+
+                        except Exception as e:
+                            self.log.debug(f'error: {e}')
+                            self.log.debug(
```

Comment:
These should be at the `log.error` level so we see errors without needing to be in debug mode.